### ISSUE TITLE
Why does `/etc/zshrc` hardcode `HISTFILE`?

### ISSUE THREAD

**Issue by il-k** [LnL7/nix-darwin#983]
Why is generated `/etc/zshrc` opinionated on this? Why can't I opt out?

**Comment by bestlem** [comment]
The reason will be that it was copied from nixos at some time e.g. https://github.com/NixOS/nixpkgs/blob/9d4bc7900334e57c09d90f985a633b2a41025129/nixos/modules/programs/zsh/zsh.nix in 2018
The nixos version now  has had some things moved to options but that has not been copied to nix-darwin. I don't know if that will fix all your issues.

**Comment by coinslinger** [comment]
Given that zsh is the primary shell out of the box with macos these days and many devs have built their macos workstream around it, its not realistic to expect to completely overwrite the root zshrc and lock it down. Breaks a lot of other tooling on the system.

Going to have to give up on nix at this point but otherwise a promising tool.

**Comment by thowar2** [comment]
Given that zsh is the primary shell out of the box with macos these days and many devs have built their macos workstream around it, its not realistic to expect to completely overwrite the root zshrc and lock it down. Breaks a lot of other tooling on the system.

Going to have to give up on nix at this point but otherwise a promising tool.

**Comment by bew** [comment]
> Breaks a lot of other tooling on the system.

How does those break?
The shell is fully controlled by the user and its config may change anything (kinda like Nix does), do you have tools that depends on specific zsh config ?

**Comment by thowar2** [comment]
> > Breaks a lot of other tooling on the system.
> 
> How does those break? The shell is fully controlled by the user and its config may change anything (kinda like Nix does), do you have tools that depends on specific zsh config ?

Yes - as with most macOS users, my entire dev workflow is configured around zsh including various zsh plugins and extensions. The terminal appearance is partially defined in zshrc. Most of the tools I install configure their PATH in zshrc.

This script just comes in and completely deletes everything, overwriting years of configurations that have built up in the zshrc and breaks how my terminal appears and functions.

Its quite impolite to just overwrite that config entirely with no effort to merge, append or work nicely with other settings or preferences.

**Comment by Samasaur1** [comment]
nix-darwin shouldn't overwrite your existing `/etc/zshrc`, it shouldn' instead exit with an error telling you to move the appropriate file out of the way. If you're sure it was us, please correct me, because that's something we should be fixing.

This is not to say that some other tool in the Nix ecosystem didn't overwrite your config, but I unfortunately only have so much time in the day :(.

Re: the root issue, yes, our module was copied from NixOS at some point and is no longer in sync. We'd be open to PRs making these settings configurable (and I'm going to go review the linked PR next) though I will note that I think you can overwrite all of the options listed in this issue using `programs.zsh.interactiveShellInit` since it is later in `/etc/zshrc`
